packages/snap/tests/e2e/createVerifiablePresentation.spec.ts

```diff
@@ -0,0 +1,249 @@
+import { AvailableMethods, ProofOptions } from '@blockchain-lab-um/masca-types';
+import { isError, Result } from '@blockchain-lab-um/utils';
+import { MetaMaskInpageProvider } from '@metamask/providers';
+import type { Json, SnapsGlobalObject } from '@metamask/snaps-types';
+import { VerifiablePresentation } from '@veramo/core';
+
+import { onRpcRequest } from '../../src';
+import { getAgent, type Agent } from '../../src/veramo/setup';
+import { account } from '../data/constants';
+import { getDefaultSnapState } from '../data/defaultSnapState';
+import exampleVCEIP712 from '../data/verifiable-credentials/exampleEIP712.json';
+import exampleVCJSONLD from '../data/verifiable-credentials/exampleJSONLD.json';
+import exampleVC_2 from '../data/verifiable-credentials/exampleJWT_2.json';
+import exampleVC_3 from '../data/verifiable-credentials/exampleJWT_3.json';
+import exampleVC from '../data/verifiable-credentials/exampleJWT.json';
+import { createMockSnap, SnapMock } from '../helpers/snapMock';
+
+const methods: AvailableMethods[] = ['did:key', 'did:jwk'];
+const proofFormats = ['jwt', /* 'lds', */ 'EthereumEip712Signature2021'];
+const proofTypes: Record<string, string> = {
+  jwt: 'JwtProof2020',
+  lds: 'Ed25519Signature2018',
+  EthereumEip712Signature2021: 'EthereumEip712Signature2021',
+};
+const options: (ProofOptions | undefined)[] = [
+  {},
+  { domain: 'localhost' },
+  { challenge: 'test-challenge' },
+  { domain: 'localhost-domain', challenge: 'challenge & domain' },
+];
+
+const vcs = [
+  { title: 'JWT', vcs: [exampleVC] },
+  { title: 'JSON-LD', vcs: [exampleVCJSONLD] },
+  { title: 'EIP712', vcs: [exampleVCEIP712] },
+  { title: '2 JWTs', vcs: [exampleVC, exampleVC_2] },
+  { title: '3 JWTs', vcs: [exampleVC, exampleVC_2, exampleVC_3] },
+  { title: 'JWT & EIP712', vcs: [exampleVC, exampleVCEIP712] },
+  { title: 'JWT & JSON-LD', vcs: [exampleVC, exampleVCJSONLD] },
+  { title: 'JSON-LD & EIP712', vcs: [exampleVCJSONLD, exampleVCEIP712] },
+];
+
+describe.each(methods)(
+  'Create VerifiablePresentation with method %s',
+  (method) => {
+    let snapMock: SnapsGlobalObject & SnapMock;
+    let issuer: string;
+    let agent: Agent;
+
+    beforeAll(async () => {
+      snapMock = createMockSnap();
+      snapMock.rpcMocks.snap_manageState({
+        operation: 'update',
+        newState: getDefaultSnapState(account),
+      });
+      const ethereumMock = snapMock as unknown as MetaMaskInpageProvider;
+      agent = await getAgent(snapMock, ethereumMock);
+      global.snap = snapMock;
+      global.ethereum = snapMock as unknown as MetaMaskInpageProvider;
+
+      snapMock.rpcMocks.snap_dialog.mockReturnValue(true);
+    });
+
+    beforeEach(async () => {
+      await agent.clear({ options: { store: ['snap', 'ceramic'] } });
+    });
+
+    it('Should switch DID method', async () => {
```

Comment:
Move this to a separate file, because its a different RPC method (not createVerifiablePresentation)